saos-persistence/src/main/java/pl/edu/icm/saos/persistence/search/implementor/LawJournalEntryJpqlSearchImplementor.java

```diff
@@ -0,0 +1,78 @@
+package pl.edu.icm.saos.persistence.search.implementor;
+
+import static com.google.common.collect.Maps.newHashMap;
+
+import java.util.Map;
+
+import org.springframework.stereotype.Service;
+
+import pl.edu.icm.saos.persistence.model.LawJournalEntry;
+import pl.edu.icm.saos.persistence.search.dto.LawJournalEntrySearchFilter;
+
+/**
+ * @author madryk
+ */
+@Service
+public class LawJournalEntryJpqlSearchImplementor extends AbstractJpqlSearchImplementor<LawJournalEntrySearchFilter, LawJournalEntry> {
+
+    
+    //------------------------ LOGIC --------------------------
+    
+    @Override
+    protected String createQuery(LawJournalEntrySearchFilter searchFilter) {
+        StringBuilder jpql = new StringBuilder(" select journalEntry from " + LawJournalEntry.class.getName() + " journalEntry ");
+        return appendConditions(searchFilter, jpql);
+    }
+
+    @Override
+    protected Map<String, Object> createParametersMap(LawJournalEntrySearchFilter searchFilter) {
+        Map<String, Object> map = newHashMap();
+        
+        if(searchFilter.getYear() != null) {
+            map.put("year", searchFilter.getYear());
+        }
+
+        if(searchFilter.getJournalNo() != null) {
+            map.put("journalNo", searchFilter.getJournalNo());
+        }
+
+        if(searchFilter.getEntry() != null) {
+            map.put("entry", searchFilter.getEntry());
+        }
+        
+        if(searchFilter.getText() != null) {
+            map.put("beginningText", searchFilter.getText() + '%');
+            map.put("middleText", "% " + searchFilter.getText() + '%');
```

Comment:
OK, I didn't notice this space ;-) 
Could you please add a simple comment to the line with middleText, e.g. // mind the space after the first % - we're looking only for words that begin with the searched phrase